### PR TITLE
implement order service event publishing 

### PR DESCRIPTION
**Pull request by andrsuh**

### CODE REVIEW

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/AddDetailDto.java [hunk 1]
```diff
@@ -15,6 +16,6 @@
 
     private Integer orderId; // if exists
 
-    @NotNull
-    private Integer userId;
+    @NotBlank
+    private String username;
```
Comment (AfanasievYuriy) [inline order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/AddDetailDto.java]: no userId?

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/OrderStatus.java [hunk 2]
```diff
@@ -3,8 +3,7 @@
 public enum OrderStatus {
     PENDING,
     CREATED,
-    APPROVED,
```
Comment (AfanasievYuriy) [inline order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/OrderStatus.java:6]: no "approving" stage, huh? what about order accounting staff?

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/AddDetailDto.java [hunk 3]
```diff
@@ -15,6 +16,6 @@
...
 }
```
Comment (AfanasievYuriy) [inline order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/AddDetailDto.java:21]: no userId?

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/PaymentDetailsDto.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.ToString;
+
+import javax.validation.constraints.Digits;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+@Getter
+@ToString
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
```
Comment (AfanasievYuriy) [inline order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/PaymentDetailsDto.java:14]: why do we need this?

File: order-management-service/app/src/main/java/com/microservices/ordermanagement/app/entity/OrderDetailEntity.java [hunk 5]
```diff
@@ -36,7 +39,16 @@
...
-        this.setTariffId(tariffDto.getId());
-        this.setPrice(tariffDto.getPrice());
+        Objects.requireNonNull(tariffDto);
+
+        this.setTariffId(Objects.requireNonNull(tariffDto.getId()));
+        this.setPrice(Objects.requireNonNull(tariffDto.getPrice()));
+        this.setDuration(Objects.requireNonNull(tariffDto.getDuration()));
+    }
+
+    boolean isTariffAssigned() {
```
Comment (AfanasievYuriy) [inline order-management-service/app/src/main/java/com/microservices/ordermanagement/app/entity/OrderDetailEntity.java:49]: access modifiers?

File: order-management-service/app/src/main/java/com/microservices/ordermanagement/app/kafka/consumer/LaundryManagementEventConsumer.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            switch (laundryManagementEvent.getPayloadCase()) {
+                case ORDERPROCESSEDEVENT: {
+                    OrderProcessedEvent orderProcessedEvent = laundryManagementEvent.getOrderProcessedEvent();
+                    logger.info("Received OrderProcessedEvent " + orderProcessedEvent);
+                    consumerSpan
+                            .customizer()
+                            .name("consume_order_processed_event");
+
+                    // here deserializer processing
+
```
Comment (AfanasievYuriy) [inline order-management-service/app/src/main/java/com/microservices/ordermanagement/app/kafka/consumer/LaundryManagementEventConsumer.java:53]: ? why don't you implement processing?

File: taskcoordinator/app/src/main/java/com/microservices/taskcoordinator/dto/inbound/OrderCoordinationDto.java [hunk 7]
```diff
@@ -22,4 +26,15 @@
...
+    public OrderCoordinationDto(OrderCreatedEvent orderCreatedEvent) {
+        this.orderId = orderCreatedEvent.getOrder().getOrderId();
+        this.details = orderCreatedEvent.getOrder().getDetailsList().stream()
+                .map(d -> new OrderDetailDto(
+                        d.getDetailId(),
+                        d.getWeight(),
+                        d.getDuration(),
+                        d.getOrderId()))
+                .collect(Collectors.toList());
+    }
```
Comment (AfanasievYuriy) [inline taskcoordinator/app/src/main/java/com/microservices/taskcoordinator/dto/inbound/OrderCoordinationDto.java:40]: like

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/OrderStatus.java (hunk 2 above)
Comment (andrsuh) [reply order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/OrderStatus.java:6]: from pending state you can wither get to created or to failed state

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/AddDetailDto.java (hunk 3 above)
Comment (andrsuh) [reply order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/AddDetailDto.java:21]: Yes, I think user can be identified by login everywhere

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/PaymentDetailsDto.java (hunk 4 above)
Comment (andrsuh) [reply order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/PaymentDetailsDto.java:14]: in order not to use setters

File: order-management-service/app/src/main/java/com/microservices/ordermanagement/app/entity/OrderDetailEntity.java (hunk 5 above)
Comment (andrsuh) [reply order-management-service/app/src/main/java/com/microservices/ordermanagement/app/entity/OrderDetailEntity.java:49]: package-private in enough

File: order-management-service/app/src/main/java/com/microservices/ordermanagement/app/kafka/consumer/LaundryManagementEventConsumer.java (hunk 6 above)
Comment (andrsuh) [reply order-management-service/app/src/main/java/com/microservices/ordermanagement/app/kafka/consumer/LaundryManagementEventConsumer.java:53]: left it to another PR

File: order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/PaymentDetailsDto.java (hunk 4 above)
Comment (AfanasievYuriy) [reply order-management-service/api/src/main/java/com/microservices/ordermanagement/api/dto/PaymentDetailsDto.java:14]: ooh
looks like your immutability dream is finally coming true))

**Review by AfanasievYuriy — DISMISSED** [review]

**Review by AfanasievYuriy — APPROVED** [review]

**Pull request merged**